# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../hfsplusutils/getloadavg.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -urN diskdev_cmds-332.14/newfs_hfs.tproj/makehfs.c diskdev_cmds-332.14-fixed/newfs_hfs.tproj/makehfs.c
--- diskdev_cmds-332.14/newfs_hfs.tproj/makehfs.c	2007-11-18 10:52:40.000000000 +0100
+++ diskdev_cmds-332.14-fixed/newfs_hfs.tproj/makehfs.c	2007-11-18 10:54:31.000000000 +0100
@@ -2120,10 +2120,11 @@
 		SHA1_Update(&context, sysctlstring, datalen);
 #endif
 		/* The system's load average: */
+#if ! defined (__UCLIBC__) && ! defined (__dietlibc__)
 		datalen = sizeof(sysloadavg);
 		getloadavg(sysloadavg, 3);
 		SHA1_Update(&context, &sysloadavg, datalen);
-
+#endif
 		/* The system's VM statistics: */
 #if !LINUX
 		mib[0] = CTL_VM;
